python/ray/workflow/tests/conftest.py

```diff
@@ -65,6 +64,38 @@ def workflow_start_regular_shared(storage_type, use_ray_client: str, request):
     ) as res:
         yield res
 
+@contextmanager
+def _workflow_start_serve(storage_url, shared, **kwargs):
+    with _init_cluster(storage_url, **kwargs) as cluster:
+        address_info = ray.init(address=cluster.address)
+        ray.workflow.init()
+        ray.serve.start(detached=True)
+        yield address_info
+
+        # The code after the yield will run as teardown code.
+        ray.serve.shutdown()
+        ray.shutdown()
+
+@pytest.fixture(scope="module")
+def workflow_start_regular_shared_serve(storage_type, request):
+    param = getattr(request, "param", {})
+    with simulate_storage(storage_type) as storage_url, _workflow_start_serve(
+        storage_url, True, **param
+    ) as res:
+        yield res
```

Comment:
Sorry, for the last minute  comment, but should we just use workflow_start ? I remember  now serve will start by itself in the code.

Reply:
If not, we should use `with _workflow_start` instead of `with _init_cluster` to cover ray client use cases.

Reply:
Fixture updated to `with _workflow_start`.